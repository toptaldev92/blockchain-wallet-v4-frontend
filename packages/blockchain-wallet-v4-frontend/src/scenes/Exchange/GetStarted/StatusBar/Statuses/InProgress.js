import { actions } from 'data'
import { Button, Text } from 'blockchain-info-components'
import { connect } from 'react-redux'
import { FormattedMessage } from 'react-intl'
import React from 'react'
import styled from 'styled-components'

const ActionButton = styled(Button).attrs({ nature: 'primary' })`
  font-weight: 600;
  width: 264px;
  height: 48px;
`

export const InProgress = ({ verifyIdentity }) => (
  <ActionButton onClick={verifyIdentity}>
    <Text color='white' size='16px' weight={600}>
      <FormattedMessage
        id='scenes.exchange.getstarted.status.inprogress.button1'
        defaultMessage='Continue Where You Left Off'
      />
    </Text>
  </ActionButton>
)

const mapDispatchToProps = dispatch => ({
  verifyIdentity: () =>
    dispatch(actions.components.identityVerification.verifyIdentity())
})

export default connect(
  undefined,
  mapDispatchToProps
)(InProgress)
