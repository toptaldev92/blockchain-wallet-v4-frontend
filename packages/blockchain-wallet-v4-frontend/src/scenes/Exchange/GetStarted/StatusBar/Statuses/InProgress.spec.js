import { InProgress } from './InProgress'
import { shallow } from 'enzyme'
import React from 'react'
import toJson from 'enzyme-to-json'

jest.mock('blockchain-info-components', () => ({
  Button: 'button'
}))

describe('InProgress', () => {
  it('renders correctly', () => {
    const component = shallow(<InProgress />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
