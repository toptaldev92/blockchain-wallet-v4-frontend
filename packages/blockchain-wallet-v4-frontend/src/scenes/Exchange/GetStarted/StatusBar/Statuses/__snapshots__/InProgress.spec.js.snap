// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InProgress renders correctly 1`] = `
<styled.button>
  <Component
    color="white"
    size="16px"
    weight={600}
  >
    <FormattedMessage
      defaultMessage="Continue Where You Left Off"
      id="scenes.exchange.getstarted.status.inprogress.button1"
      values={Object {}}
    />
  </Component>
</styled.button>
`;
